### DISCUSSION TITLE
Managing a player or npcs within the tilemap

### DISCUSSION THREAD

**Discussion by rwygand** [StarArawn/bevy_ecs_tilemap#424] [General]
Hi, new to bevy and rust and all sorts of things. One thing I'm confused about and isn't covered very well in the examples, AFAICT, is how to spawn and manage other entities that occupy the tilemap.

If I'm building an outdoor map, with two tag structs, Field{} and Wall{} and I want to spawn a player at a valid position on the tilemap, I get that I could pick a random tile of all the Field{} types, but then do I insert the player's texture directly into the tilemap at that position? Would I have a second layer for tracking moveables? or should I be maintaining another data structure with positions of all mobs?

The old bevy_tileset crate has an example of creating a random map with player movement, and I think this would be a great example to help newbs like myself get started.

**Comment by ifbsj** [comment]
The best choice is probably to create your player as a regular Bevy SpriteBundle.
Only if you are doing something like a chess game, where the player always moves from one tile to the next, then you could make the player on a separate tile layer with higher z value (so it is drawn on top of the other tilemap).
